Add support for f2fs external storage, Adds f2fs-tools package & f2fs support to kernel/scripts

aenertia opened the pull request:
Make kernel CONFIG options consistent for all targets as related to F2FS support.

  This adds new f2fs-tools package for supporting f2fs filesystems and required changes to automount scripts
  Tested: RGB30, RG353M

    
        * Add f2fs support in automount script
        + modify udevil to remove compression mount option for btrfs which may
          be cause of queue load
        - add f2fs-tools to virtual package

porschemad911 commented on packages/sysutils/f2fs-tools/package.mk:
```diff
@@ -0,0 +1,12 @@
+PKG_SITE="https://git.kernel.org/pub/scm/linux/kernel/git/jaegeuk/f2fs-tools.git/"
+PKG_URL="${PKG_SITE}/snapshot/${PKG_NAME}-${PKG_VERSION}.tar.gz"
+PKG_DEPENDS_TARGET="toolchain libtool e2fsprogs util-linux zlib systemd lzo"
+PKG_LONGDESC="NAND flash memory-based storage devices, such as SSD, and SD cards, have been widely being used for ranging from mobile to server systems. Since they are known to have different characteristics from the conventional rotational disks,a file system, an upper layer to the storage device, should adapt to the changes from the sketch. F2FS is a new file system carefully designed for the N ...
```
TLDR, maybe change to `F2FS is a file system carefully designed for the NAND flash memory-based storage devices.`

aenertia replied:
Heh I stole this text from the maintainers description verbatim

aenertia replied:
am just waiting to ensure builds for all targets work ok and will update this PR with the automount and kernel bits needed to enable this

porschemad911 commented on packages/sysutils/f2fs-tools/package.mk:
```diff
@@ -0,0 +1,12 @@
+PKG_NAME="f2fs-tools"
+PKG_VERSION="1.16.0"
+PKG_LICENSE="GPL"
+PKG_SITE="https://git.kernel.org/pub/scm/linux/kernel/git/jaegeuk/f2fs-tools.git/"
```
This isn't used in the build process right? Update this to be a URL it makes sense to visit, not the `.git` URL.

aenertia commented:
Actually it is; The only other reference would be a clone of the kernel.org
repo on github. It's rendered with an about/html etc so I would have
thought this is the correct reference for the package rather than the
github clone.

On Mon, 6 Jan 2025 at 18:16, John Williams ***@***.***> wrote:

> ***@***.**** commented on this pull request.
> ------------------------------
>
> In packages/sysutils/f2fs-tools/package.mk
> <https://github.com/ROCKNIX/distribution/pull/1089#discussion_r1903662528>
> :
>
> > @@ -0,0 +1,12 @@
> +# SPDX-License-Identifier: GPL-2.0
> +# Copyright (C) 2024-present ROCKNIX (https://github.com/ROCKNIX)
> +
> +
> +PKG_NAME="f2fs-tools"
> +PKG_VERSION="1.16.0"
> +PKG_LICENSE="GPL"
> +PKG_SITE="https://git.kernel.org/pub/scm/linux/kernel/git/jaegeuk/f2fs-tools.git/"
>
> This isn't used in the build process right? Update this to be a URL it
> makes sense to visit, not the .git URL.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ROCKNIX/distribution/pull/1089#pullrequestreview-2531340479>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AACF5L56BF6E4T2J7XSI3S32JIGUBAVCNFSM6AAAAABUUX6JCWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKMZRGM2DANBXHE>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

aenertia committed:
Make kernel CONFIG options consistent for all targets as related to F2FS support

    * Add f2fs support in automount script
    + modify udevil to remove compression mount option for btrfs which may
      be cause of queue load
    - add f2fs-tools to virtual package

This adds new f2fs-tools package for supportting f2fs filesystems

tested: RGB30, RG353M

porschemad911 commented:
> Actually it is; The only other reference would be a clone of the kernel.org repo on github. It's rendered with an about/html etc so I would have thought this is the correct reference for the package rather than the github clone.
> […](#)
> On Mon, 6 Jan 2025 at 18:16, John Williams ***@***.***> wrote: ***@***.**** commented on this pull request. ------------------------------ In packages/sysutils/f2fs-tools/package.mk <[#1089 (comment)](https://github.com/ROCKNIX/distribution/pull/1089#discussion_r1903662528)> : > @@ -0,0 +1,12 @@ +# SPDX-License-Identifier: GPL-2.0 +# Copyright (C) 2024-present ROCKNIX (https://github.com/ROCKNIX) + + +PKG_NAME="f2fs-tools" +PKG_VERSION="1.16.0" +PKG_LICENSE="GPL" +PKG_SITE="https://git.kernel.org/pub/scm/linux/kernel/git/jaegeuk/f2fs-tools.git/" This isn't used in the build process right? Update this to be a URL it makes sense to visit, not the .git URL. — Reply to this email directly, view it on GitHub <[#1089 (review)](https://github.com/ROCKNIX/distribution/pull/1089#pullrequestreview-2531340479)>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AACF5L56BF6E4T2J7XSI3S32JIGUBAVCNFSM6AAAAABUUX6JCWVHI2DSMVQWIX3LMV43YUDVNRWFEZLROVSXG5CSMV3GSZLXHMZDKMZRGM2DANBXHE> . You are receiving this because you authored the thread.Message ID: ***@***.***>

All good, I realised and removed my comment earlier.

r3claimer commented:
Closing due to inactivity and if still important need to rebase.
